# E-COMMERCE BUSINESS TRANSACTION ANALYSIS

## This analysis is based on the sales transaction data set out of a UK-based B2C e-commerce company (online retail) during the period of one year.

#### E-commerce has become a new channel to support businesses development. Through e-commerce, businesses can get access and establish a wider market presence by providing cheaper and more efficient distribution channels for their products or services. E-commerce has also changed the way people shop and consume products and services. Many people are turning to their computers or smart devices to order goods, which can easily be delivered to their homes.

We have to import the libraries that we'll use to handle our data.

In [1]:
import pandas as pd
import numpy as np

We import the dataset that contains all the data for our analysis.

In [2]:
df = pd.read_csv("ecommerce.csv")

Now we have to check the information contained into our dataset.

In [3]:
df.head()

TransactionNo       Date ProductNo                          ProductName  \
0        581482  12/9/2019     22485        Set Of 2 Wooden Market Crates   
1        581475  12/9/2019     22596  Christmas Star Wish List Chalkboard   
2        581475  12/9/2019     23235             Storage Tin Vintage Leaf   
3        581475  12/9/2019     23272    Tree T-Light Holder Willie Winkie   
4        581475  12/9/2019     23239    Set Of 4 Knick Knack Tins Poppies   

   Price  Quantity  CustomerNo         Country  
0  21.47        12     17490.0  United Kingdom  
1  10.65        36     13069.0  United Kingdom  
2  11.53        12     13069.0  United Kingdom  
3  10.65        12     13069.0  United Kingdom  
4  11.94         6     13069.0  United Kingdom

In [4]:
df.tail()

TransactionNo       Date ProductNo                   ProductName  \
536345       C536548  12/1/2018     22168  Organiser Wood Antique White   
536346       C536548  12/1/2018     21218        Red Spotty Biscuit Tin   
536347       C536548  12/1/2018     20957  Porcelain Hanging Bell Small   
536348       C536548  12/1/2018     22580  Advent Calendar Gingham Sack   
536349       C536548  12/1/2018     22767    Triple Photo Frame Cornice   

        Price  Quantity  CustomerNo  Country  
536345  18.96        -2     12472.0  Germany  
536346  14.09        -3     12472.0  Germany  
536347  11.74        -1     12472.0  Germany  
536348  16.35        -4     12472.0  Germany  
536349  20.45        -2     12472.0  Germany

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 536350 entries, 0 to 536349
Data columns (total 8 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   TransactionNo  536350 non-null  object 
 1   Date           536350 non-null  object 
 2   ProductNo      536350 non-null  object 
 3   ProductName    536350 non-null  object 
 4   Price          536350 non-null  float64
 5   Quantity       536350 non-null  int64  
 6   CustomerNo     536295 non-null  float64
 7   Country        536350 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 22.5+ MB


##### Our dataset has 8 columns, the description of their content is as follow:

1. TransactionNo (categorical): a six-digit unique number that defines each transaction. The letter “C” in the code indicates a cancellation.

2. Date (numeric): the date when each transaction was generated.

3. ProductNo (categorical): a five or six-digit unique character used to identify a specific product.

4. Product (categorical): product/item name.

5. Price (numeric): the price of each product per unit in pound sterling (£).

6. Quantity (numeric): the quantity of each product per transaction. Negative values related to cancelled transactions.

7. CustomerNo (categorical): a five-digit unique number that defines each customer.

8. Country (categorical): name of the country where the customer resides.

##### DISCLAIMER: 

```
For the correct interpretation of this analysis we have to be aware that there is a small percentage of order cancellation in the data set. These cancellations were due to out-of-stock conditions on some products. Under this situation, customers tend to cancel an order as they want all products delivered all at once. These cancellations are represented in negative numbers in the 'Quantity' column.
```

The first thing that we're going to check is if there are people interested in the products offered, and since this is an online retail that sells worlwide, we're also interested in knowing where all these sells are coming from. Here we can see all the countries where this e-commerce's customers are from:

In [6]:
print(df.groupby('Country').size())

Country
Australia                 1704
Austria                    887
Bahrain                     17
Belgium                   2539
Brazil                      31
Canada                     150
Channel Islands            629
Cyprus                     582
Czech Republic              28
Denmark                    416
EIRE                      8048
European Community          58
Finland                    692
France                   10526
Germany                  10675
Greece                      67
Hong Kong                  153
Iceland                    787
Israel                     358
Italy                      661
Japan                      371
Lebanon                     45
Lithuania                   34
Malta                      149
Netherlands               2330
Norway                     938
Poland                     174
Portugal                  1848
RSA                         57
Saudi Arabia                10
Singapore                  215
Spain                     2430


It's not a surprise to see that a huge majority of customers reside in UK (485095), followed by those in other countries from Europe like Germany (10675), France (10526), Ireland (8048), Spain (2430), Netherlands (2330) or Switzerland (2336).

Now we want to know what's the average price that customers have paid in each country where this e-commerce made a transaction:

In [7]:
ctry_mean = df.groupby('Country').mean()
ctry_mean = ctry_mean[["Price"]]
print(ctry_mean)

                          Price
Country                        
Australia             12.774536
Austria               13.733980
Bahrain               15.010000
Belgium               13.185282
Brazil                14.382903
Canada                12.348467
Channel Islands       13.585199
Cyprus                19.033368
Czech Republic        11.866786
Denmark               12.472885
EIRE                  13.037793
European Community    14.597241
Finland               12.617977
France                12.422292
Germany               12.596319
Greece                11.540149
Hong Kong             13.394183
Iceland                7.867891
Israel                13.315335
Italy                 13.323510
Japan                 12.096092
Lebanon               15.771111
Lithuania             13.066471
Malta                 13.580336
Netherlands           11.888781
Norway                12.611418
Poland                13.888333
Portugal              11.986640
RSA                   14.070175
Saudi Ar

In [8]:
ctry_mean.to_csv('ctry_mean.csv')

Then we'll look for the minimum price customers paid in each country:

In [9]:
ctry_min = df.groupby('Country').min()
ctry_min = ctry_min[['Price']]
print(ctry_min)

                      Price
Country                    
Australia              5.13
Austria                5.55
Bahrain               11.53
Belgium                5.97
Brazil                 5.97
Canada                 5.97
Channel Islands        5.97
Cyprus                 5.97
Czech Republic        10.55
Denmark                5.46
EIRE                   5.46
European Community    10.81
Finland                5.97
France                 5.46
Germany                5.34
Greece                 6.04
Hong Kong              5.97
Iceland                5.13
Israel                 5.97
Italy                  5.97
Japan                  5.87
Lebanon               10.81
Lithuania             11.53
Malta                  5.97
Netherlands            5.55
Norway                 6.04
Poland                10.65
Portugal               5.55
RSA                    5.97
Saudi Arabia          10.68
Singapore              6.13
Spain                  5.46
Sweden                 5.46
Switzerland         

In [10]:
ctry_min.min()

Price    5.13
dtype: float64

In [11]:
ctry_min.to_csv('ctry_min.csv')

As well as the maximum prices customers have paid in each country:

In [12]:
ctry_max = df.groupby('Country').max()
ctry_max = ctry_max[['Price']]
print(ctry_max)

                       Price
Country                     
Australia              44.37
Austria                40.11
Bahrain                23.32
Belgium                51.20
Brazil                 21.47
Canada                 23.32
Channel Islands        35.82
Cyprus                660.62
Czech Republic         15.32
Denmark                27.62
EIRE                  179.37
European Community     25.57
Finland                35.82
France                 54.27
Germany                70.00
Greece                 25.57
Hong Kong              27.62
Iceland                23.32
Israel                138.38
Italy                  51.20
Japan                  25.57
Lebanon                25.57
Lithuania              16.35
Malta                  26.60
Netherlands           123.00
Norway                 35.82
Poland                 51.20
Portugal               40.95
RSA                    25.57
Saudi Arabia           15.88
Singapore             660.62
Spain                  76.88
Sweden        

In [13]:
ctry_max.max()

Price    660.62
dtype: float64

In [14]:
ctry_max.to_csv('ctry_max.csv')

Now we just have left to look for the highest quantity ever sold at once.

In [15]:
qty_max = df.groupby('ProductName').max()
qty_max = qty_max[['Quantity']]
print(qty_max)

                                  Quantity
ProductName                               
10 Colour Spaceboy Pen                 288
12 Coloured Party Balloons             100
12 Daisy Pegs In Wood Box               24
12 Egg House Painted Wood                8
12 Hanging Eggs Hand Painted            10
...                                    ...
Zinc T-Light Holder Stars Small        216
Zinc Top 2 Door Wooden Shelf             2
Zinc Willie Winkie Candle Stick        192
Zinc Wire Kitchen Organiser              4
Zinc Wire Sweetheart Letter Tray        24

[3768 rows x 1 columns]


In [16]:
qty_max.describe()

Quantity
count   3768.000000
mean     174.051752
std     1817.804416
min     -200.000000
25%       12.000000
50%       48.000000
75%      144.000000
max    80995.000000

In [17]:
qty_max.to_csv('qty_max.csv')

# CONCLUSION

## What are the top 3 countries where customers buy from?

The top 3 countries where customers buy from are:
1. UK.
2. Germany.
3. France.

## What's the minimum and the maximum price that customers from these countries paid?

When buying products, customers paid a minimum price of £5.13, as well as a maximum price of £660.62 overall the countries.
In UK the minimum price the paid was £5.13 and the maximum price £660.62. In Germany the minimum price the paid was £5.34 and the maximum price £70.00. In France the minimum price the paid was £5.46 and the maximum price £54.27. 

## What was the average quantity, as well as the minimum and the maximum quantities that customers bought at once?

Customers buy on average 174 units of a single product at once (many of these buyers are small businesses who buy to resell on their own), the minimum units ever sold at once was -200 (it means that a purchase of 200 units was cancelled) and the maximum ever sold was 80995.